app/views/spree/shared/_payment.html.haml

```diff
@@ -13,3 +13,9 @@
     = t :email_payment_summary
 - if @order.payments.any?
   = render partial: 'spree/shared/payments_list', locals: { payments: @order.payments }
+  - completedPayments = @order.payments.select {|p| p.state == "checkout" || p.state == "completed"}
```

Comment:
IMO this is too much business logic to stuff in the view. There are a few options to solve it but to me, the best is to pass on the data to the view as we do with the rest. Through an instance variable so that the view is 100% decoupled from the actual payments.

Then, the other issue I see is that this concept of completed payments is surely duplicate. It's something used throughout the app and so it'd be best to reuse it. If you `git grep` and go through controllers I'm sure you'll find it.

Reply:
That's my tropism as a front developper. I put a lot of logic inside my views (but when I use React, ... I can easily separate between components, containers, ... which comes to the same thing.). I have to get used to it.

Reply:
I certainly could use the already existing `PaymentMethodsHelper`

Reply:
but I want to stress out that what in my experience leads to decoupled views (hence a better world) is to have something like the following in the controller:

```rb
def whatever_action
  (...)
  @order = ...
  @complete_payment = WhateverClassThatReturnsThePaymentsWeWant.new.call.first
end
```

then the view becomes trivial:

```rb
- if @complete_payment.present?
  %p{style: "margin: 5px"}=  payment_method(@complete_payment).description
```

Reply:
Yes, I totally agree.

Reply:
I guess this is a more elegant way of doing with commit: 843877a0b8d201611b4e80ce4b35e99e22973a0d